docs/docs/50-user-guide/60-reference-docs/40-expressions.md

```diff
@@ -740,6 +740,66 @@ config:
   status: ${{ status("my-step-alias") }}
 ```
 
+### `semverParse(version)`
+
+The `semverParse()` function parses a semantic version string and returns a
+struct that provides access to its individual components through method calls.
+This is useful for scenarios where you need to inspect or manipulate specific
+version components, such as bumping the major, minor, or patch version.
+
+The returned struct provides the following methods:
+
+| Method | Return Type | Description |
+|--------|-------------|-------------|
+| `Major()` | `number` | Returns the major version component. |
+| `Minor()` | `number` | Returns the minor version component. |
+| `Patch()` | `number` | Returns the patch version component. |
+| `Prerelease()` | `string` | Returns the prerelease identifier (empty string if none). |
+| `Metadata()` | `string` | Returns the build metadata (empty string if none). |
+| `IncMajor()` | `Version` | Returns a new version with the major version incremented and minor/patch reset to 0. |
+| `IncMinor()` | `Version` | Returns a new version with the minor version incremented and patch reset to 0. |
+| `IncPatch()` | `Version` | Returns a new version with the patch version incremented. |
+| `String()` | `string` | Returns the full version string. |
+
+:::info
+The function uses the [Semantic Versioning](https://semver.org/) specification
+to parse versions. It supports versions with or without the `v` prefix, as well
+as prerelease and build metadata components.
+:::
+
+Example:
+
+```yaml
+# Read current chart version and bump the minor version
+
+- uses: yaml-parse
+  as: read-version
+  config:
+    path: ./Chart.yaml
+    outputs:
+    - name: currentVersion
+      fromExpression: version
+    - name: parsed
+      fromExpression: semverParse(version)
+
+# Update chart with bumped minor version (e.g., 1.2.3 → 1.3.0)
+- uses: yaml-update
+  config:
+    path: ./Chart.yaml
+    updates:
+    - key: version
+      value: ${{ task.outputs['read-version'].parsed.IncMinor().String() }}
+
+# Or access individual components
+- uses: set-var
+  config:
+    vars:
+    - name: majorVersion
+      value: ${{ task.outputs['read-version'].parsed.Major() }}
+    - name: minorVersion
+      value: ${{ task.outputs['read-version'].parsed.Minor() }}
```

Comment:
`set-var` isn't a step type that exists. I would just drop this example entirely because the one above is (after corrections) as great an example of realistic use as I can imagine. I don't think we need this extra bit.

```suggestion
```